staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -310,6 +310,19 @@ var (
 		},
 		[]string{priorityLevel, "success"},
 	)
+	apiserverWorkEstimateSeats = compbasemetrics.NewHistogramVec(
+		&compbasemetrics.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "work_estimate_seats_samples",
```

Comment:
`work_estimated_seats` might be a better name since we want the unit as a suffix.

Also, I am wondering if this metric should really expose the notion of `seats`. As far as I can tell this is a very low-level implementation detail that will most likely not be understandable to non P&F experts that might be interested in this metric. So maybe we should mention concurrency instead of seats? Something like `request_estimated_concurrency` may be easier to understand, wydt?

Reply:
I am somewhat sympathetic to the naming concern, but do not think "concurrency" solves the problem.  If I did not understand APF and saw a metric named `request_estimated_concurrency` I would assume this is an estimate of how many concurrent goroutines will be used to serve the request --- which is not what we mean.  Perhaps "width" or "weight"?  Or something wordier but I think quite on point: "cost factor".

Reply:
I am not familiar enough with the implementation details to weigh in, maybe @tkashem and @wojtek-t have some suggestions.

Besides that, the more I think about it, the more I am wondering if this metric would really be useful to the average k8s user. I am not saying that because I think that this metric has no value but rather because I think that the target of this kind of low-level metric is the Kubernetes developers rather than the Kubernetes users. This distinction is important here since the average k8s users might not be interested in the same metrics as someone who knows the intricate details of how k8s works.
We recently had this idea in sig-instrumentation to add `debug` level metrics that would only be meaningful to the developers of k/k and as such would be disabled by default since they wouldn't make sense to average users. This doesn't exist yet, but I am thinking that this metric among some others might be a good fit for this kind of improvement.
Going back to the original naming discussion having debug level metrics would allow us to stick to the original name that is referencing low-level implementation details since the target of this metric would be the developers of this feature only. Let me know what are your thoughts on that, maybe I am missing something.

Reply:
This is a metric that will be helpful in diagnosing situations --- which we have found in testing and may well occur in the wild --- where APF allows too few requests to be active at a time.

If it is explained to operators that debug level metrics can have very helpful information when things go wrong, then this could be a debug level metric.

Reply:
I didn't know how it could be used in practice, but after your explanation, it makes more sense. Considering that it would help to diagnose issues, then it makes sense to have it available to everyone.

The idea behind debug metrics was more to enable k/k developers to create ephemeral metrics that would help them during their development process.

Reply:
@dgrisonnet i renamed the metric to `work_estimated_seats`